File: Problems/Algorithms/1641. Count Sorted Vowel Strings/count_sorted_vowel_strings.py
class Solution:
    def countVowelStrings(self, n: int) -> int:
        a, e, i, o, u = 1, 1, 1, 1, 1

        for _ in range(n-1):
            a += e + i + o + u
            e += i + o + u
            i += o + u
            o += u

        return a + e + i + o + u

File: Problems/Algorithms/1641. Count Sorted Vowel Strings/CountSortedVowelStrings.kt
class Solution {
    private val curr: StringBuilder
    private var res = 0
    private val vowels: CharArray = charArrayOf('a', 'e', 'i', 'o', 'u')
    
    init {
        curr = StringBuilder()
    }
    
    fun countVowelStrings(n: Int): Int {
        aux(n, 0)
        return res
    }
    
    private fun aux(n: Int, start: Int): Unit {
        if (curr.length == n) {
            res++
            return
        }
        
        for (i in start..4) {
            curr.append(vowels[i])
            aux(n, i)
            curr.deleteCharAt(curr.length-1)
        }
    }
}

